tensorboard/plugins/pr_curve/pr_curve_plugin_test.py

```diff
@@ -0,0 +1,117 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests the op that generates pr_curve summaries."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import json
+import os
+
+import numpy as np
+import tensorflow as tf
+
+from google.protobuf import json_format
+from tensorboard.backend.event_processing import event_multiplexer
+from tensorboard.plugins.pr_curve import summary
+from tensorboard.plugins.pr_curve import pr_curve_pb2
+
+
+class PrCurveTest(tf.test.TestCase):
+
+  def setUp(self):
+    self.logdir = self.get_temp_dir()
+    tf.reset_default_graph()
+
+  def test1Class(self):
+    # Generate summaries for showing PR curves in TensorBoard.
+    with tf.Session() as sess:
+      summary.op(
+          tag='tag_bar',
+          labels=tf.constant([True, False, True, False], dtype=tf.bool),
+          predictions=tf.constant([0.8, 0.6, 0.4, 0.2], dtype=tf.float32),
+          num_thresholds=10)
+      merged_summary_op = tf.summary.merge_all()
+      foo_directory = os.path.join(self.logdir, 'foo')
+      writer = tf.summary.FileWriter(foo_directory, sess.graph)
+      writer.add_summary(sess.run(merged_summary_op), 1)
+      writer.close()
+
+    # Create a multiplexer for reading the data we just wrote.
+    multiplexer = event_multiplexer.EventMultiplexer()
+    multiplexer.AddRunsFromDirectory(self.logdir)
+    multiplexer.Reload()
+
+    # Verify that the metadata was correctly written.
+    accumulator = multiplexer.GetAccumulator('foo')
+    tag_content_dict = accumulator.PluginTagToContent('pr_curve')
+    self.assertListEqual(['tag_bar/tag_bar'], list(tag_content_dict.keys()))
+
+    # Parse the data within the JSON string and set the proto's fields.
+    plugin_data = pr_curve_pb2.PrCurvePluginData()
+    json_format.Parse(tag_content_dict['tag_bar/tag_bar'], plugin_data)
+    self.assertEqual(10, plugin_data.num_thresholds)
+
+    # Test the summary contents.
+    tensorEvents = accumulator.Tensors('tag_bar/tag_bar')
+    self.assertEqual(1, len(tensorEvents))
+    tensorEvent = tensorEvents[0]
+    self.assertEqual(1, tensorEvent.step)
+    tensorProto = tensorEvent.tensor_proto
+
+    # The tensor shape must be correct. The first dimension is the 4 categories
+    # of counts. The 2nd dimension is the number of classes. The last dimension
+    # is the number of thresholds.
+    self.assertEqual(3, len(tensorProto.tensor_shape.dim))
+    correct_shape = [4, 1, 10]
+    self.assertListEqual(
+        [4, 1, 10], [dim.size for dim in tensorProto.tensor_shape.dim])
+
+    # The counts within the various bins must be correct.
+    values = np.reshape(
+        np.fromstring(tensorProto.tensor_content, dtype=np.int64),
```

Comment:
I believe that it's best practice to use `tf.make_ndarray(tensor_proto)` instead of directly accessing the `tensor_content`, because either the `tensor_content` or the `int_val` could be used to store the same data. (Once you've done this, you may as well do the same for the shape checks above: treating the `tensor_proto` as opaque is not a bad idea.)